### ISSUE TITLE
Trait Error about box_style when running tutorial code

### ISSUE THREAD

**Issue by jonaskindler93** [menpo/menpowidgets#21]
Hey,

I just tried to run the code from the following Tutorial:
http://www.menpo.org/menpofit/pdm.html

specifically
`%matplotlib inline
from menpowidgets import visualize_pointclouds
visualize_pointclouds(training_shapes)`

unfortunately I get the following error:

'TraitError: The 'box_style' trait of a ColourSelectionWidget instance must be any of [u'success', u'info', u'warning', u'danger', u''], but a value of None <type 'NoneType'> was specified.'

I'm on python 2.7 and Jupyter Notebook 4.2.2

Does anybody else have the same issue?

Thanks in advance :)

**Comment by bobetocalo** [comment]
```
%matplotlib inline
import menpo.io as mio
image = mio.import_builtin_asset.einstein_jpg()
image.view_landmarks()
```

Initializing...
TraitError: The 'box_style' trait of a ColourSelectionWidget instance must be any of [u'success', u'info', u'warning', u'danger', u''], but a value of None <type 'NoneType'> was specified.

**Comment by nontas** [comment]
That is almost definitely because you're using `ipywidgets=5.*` or `ipywidgets=6.*`. Unfortunately, the current release of `menpowidgets` depends on `ipywidgets>=4.1.1,<5.0`. Could you check your `conda list`? We plan to upgrade `menpowidgets` to be compatible with the latest `ipywidgets`, but I'm not sure when this is going to happen.

**Comment by bobetocalo** [comment]
In my case, it should be something different because
ipywidgets                4.1.1                    py27_0

**Comment by nontas** [comment]
Can you paste your `conda list` output here?

**Comment by bobetocalo** [comment]
```
_license                  1.1                      py27_1  
_nb_ext_conf              0.3.0                    py27_0  
alabaster                 0.7.9                    py27_0  
anaconda                  custom                   py27_0  
anaconda-clean            1.0.0                    py27_0  
anaconda-client           1.5.4                    py27_0  
anaconda-navigator        1.3.2                    py27_0  
apptools                  4.3.0                    py27_0  
argcomplete               1.0.0                    py27_1  
astroid                   1.4.7                    py27_0  
astropy                   1.2.1               np111py27_0  
babel                     2.3.4                    py27_0  
backports                 1.0                      py27_0  
backports_abc             0.4                      py27_0  
beautifulsoup4            4.5.1                    py27_0  
bitarray                  0.8.1                    py27_0  
blaze                     0.10.1                   py27_0  
bokeh                     0.12.3                   py27_0  
boost                     1.59.0                   py27_0    menpo
boto                      2.43.0                   py27_0  
bottleneck                1.1.0               np111py27_0  
bzip2                     1.0.6                         3  
cairo                     1.12.18                       6  
cdecimal                  2.3                      py27_2  
cffi                      1.8.3                    py27_0  
chest                     0.2.3                    py27_0  
click                     6.6                      py27_0  
cloudpickle               0.2.1                    py27_0  
clyent                    1.2.2                    py27_0  
colorama                  0.3.7                    py27_0  
conda                     4.2.12                   py27_0  
conda-build               2.0.8                    py27_0  
conda-env                 2.6.0                         0  
configobj                 5.0.6                    py27_0  
configparser              3.5.0                    py27_0  
contextlib2               0.5.4                    py27_0  
cryptography              1.5.3                    py27_0  
curl                      7.49.0                        1  
cycler                    0.10.0                   py27_0  
cyffld2                   0.2.4                    py27_0    menpo
cypico                    0.2.7                    py27_0    menpo
cyrasterize               0.3.2                    py27_0    menpo
cython                    0.25.1                   py27_0  
cytoolz                   0.8.0                    py27_0  
cyvlfeat                  0.4.5                    py27_0    menpo
dask                      0.12.0                   py27_0  
datashape                 0.5.2                    py27_0  
dbus                      1.10.10                       0  
decorator                 4.0.10                   py27_0  
dill                      0.2.5                    py27_0  
dlib                      18.18                    py27_2    menpo
docopt                    0.6.2                    py27_0  
docutils                  0.12                     py27_2  
dynd-python               0.7.2                    py27_0  
entrypoints               0.2.2                    py27_0  
enum34                    1.1.6                    py27_0  
envisage                  4.5.1                    py27_0    menpo
et_xmlfile                1.0.1                    py27_0  
expat                     2.1.0                         0  
fastcache                 1.0.2                    py27_1  
ffmpeg                    2.7.0                         0    menpo
fftw                      3.3.4                         0    menpo
filelock                  2.0.7                    py27_0  
flask                     0.11.1                   py27_0  
flask-cors                2.1.2                    py27_0  
fontconfig                2.11.1                        6  
freetype                  2.5.5                         1  
funcsigs                  1.0.2                    py27_0  
functools32               3.2.3.2                  py27_0  
futures                   3.0.5                    py27_0  
get_terminal_size         1.0.0                    py27_0  
gevent                    1.1.2                    py27_0  
glew                      2.0.0                         0    menpo
glfw3                     3.2.1                         0    menpo
glib                      2.43.0                        1  
greenlet                  0.4.10                   py27_0  
grin                      1.2.1                    py27_3  
gst-plugins-base          1.8.0                         0  
gstreamer                 1.8.0                         0  
h5py                      2.6.0               np111py27_2  
harfbuzz                  0.9.39                        1  
hdf5                      1.8.17                        1  
heapdict                  1.0.0                    py27_1  
icu                       54.1                          0  
idna                      2.1                      py27_0  
imagesize                 0.7.1                    py27_0  
ipaddress                 1.0.17                   py27_0  
ipykernel                 4.5.0                    py27_0  
ipython                   4.2.0                    py27_0  
ipython_genutils          0.1.0                    py27_0  
ipywidgets                4.1.1                    py27_0  
itsdangerous              0.24                     py27_0  
jbig                      2.1                           0  
jdcal                     1.3                      py27_0  
jedi                      0.9.0                    py27_1  
jinja2                    2.8                      py27_1  
jpeg                      8d                            2  
jsonschema                2.5.1                    py27_0  
jupyter                   1.0.0                    py27_3  
jupyter_client            4.4.0                    py27_0  
jupyter_console           5.0.0                    py27_0  
jupyter_core              4.2.0                    py27_0  
lazy-object-proxy         1.2.1                    py27_0  
libdynd                   0.7.2                         0  
libffi                    3.2.1                         0  
libgcc                    5.2.0                         0  
libgfortran               3.0.0                         1  
libpng                    1.6.22                        0  
libsodium                 1.0.10                        0  
libtiff                   4.0.6                         2  
libxcb                    1.12                          1  
libxml2                   2.9.2                         0  
libxslt                   1.1.28                        0  
llvmlite                  0.14.0                   py27_0  
locket                    0.2.0                    py27_1  
lxml                      3.6.4                    py27_0  
markupsafe                0.23                     py27_2  
matplotlib                1.5.3               np111py27_1  
mayavi                    4.5.0                    py27_0    menpo
menpo                     0.7.4                    py27_0    menpo
menpo3d                   0.4.2                    py27_0    menpo
menpocli                  0.0.2                    py27_0    menpo
menpodetect               0.4.1                    py27_0    menpo
menpofit                  0.4.1                    py27_0    menpo
menpoproject              2.0                        py_0    menpo
menpowidgets              0.2.1                    py27_0    menpo
mistune                   0.7.3                    py27_0  
mkl                       11.3.3                        0  
mkl-service               1.1.2                    py27_2  
mock                      2.0.0                    py27_0  
mpmath                    0.19                     py27_1  
multipledispatch          0.4.9                    py27_0  
nb_anacondacloud          1.2.0                    py27_0  
nb_conda                  2.0.0                    py27_0  
nb_conda_kernels          2.0.0                    py27_0  
nbconvert                 4.2.0                    py27_0  
nbformat                  4.1.0                    py27_0  
nbpresent                 3.0.2                    py27_0  
networkx                  1.11                     py27_0  
nltk                      3.2.1                    py27_0  
nose                      1.3.7                    py27_1  
notebook                  4.2.3                    py27_0  
numba                     0.29.0              np111py27_0  
numexpr                   2.6.1               np111py27_1  
numpy                     1.11.2                   py27_0  
odo                       0.5.0                    py27_1  
opencv3                   3.1.0                    py27_0    menpo
openpyxl                  2.4.0                    py27_0  
openssl                   1.0.2j                        0  
pandas                    0.19.1              np111py27_0  
partd                     0.3.6                    py27_0  
patchelf                  0.9                           0  
path.py                   8.2.1                    py27_0  
pathlib                   1.0                      py27_1    menpo
pathlib2                  2.1.0                    py27_0  
patsy                     0.4.1                    py27_0  
pbr                       1.10.0                   py27_0  
pep8                      1.7.0                    py27_0  
pexpect                   4.0.1                    py27_0  
pickleshare               0.7.4                    py27_0  
pillow                    3.4.2                    py27_0  
pip                       9.0.1                    py27_0  
pixman                    0.32.6                        0  
pkginfo                   1.4.1                    py27_0  
ply                       3.9                      py27_0  
prompt_toolkit            1.0.9                    py27_0  
psutil                    5.0.0                    py27_0  
ptyprocess                0.5.1                    py27_0  
py                        1.4.31                   py27_0  
pyasn1                    0.1.9                    py27_0  
pycairo                   1.10.0                   py27_0  
pycosat                   0.6.1                    py27_1  
pycparser                 2.17                     py27_0  
pycrypto                  2.6.1                    py27_4  
pycurl                    7.43.0                   py27_0  
pyface                    4.5.2                    py27_0  
pyflakes                  1.3.0                    py27_0  
pygments                  2.1.3                    py27_0  
pylint                    1.5.4                    py27_1  
pyopenssl                 16.2.0                   py27_0  
pyparsing                 2.1.4                    py27_0  
pyqt                      5.6.0                    py27_0  
pytables                  3.3.0               np111py27_0  
pytest                    3.0.3                    py27_0  
python                    2.7.12                        1  
python-dateutil           2.6.0                    py27_0  
pytz                      2016.7                   py27_0  
pyyaml                    3.12                     py27_0  
pyzmq                     16.0.0                   py27_0  
qt                        5.6.0                         1  
qtawesome                 0.3.3                    py27_0  
qtconsole                 4.2.1                    py27_1  
qtpy                      1.1.2                    py27_0  
readline                  6.2                           2  
redis                     3.2.0                         0  
redis-py                  2.10.5                   py27_0  
requests                  2.11.1                   py27_0  
rope                      0.9.4                    py27_1  
ruamel_yaml               0.11.14                  py27_0  
scikit-image              0.12.3              np111py27_1  
scikit-learn              0.18                np111py27_0  
scipy                     0.18.1              np111py27_0  
setuptools                27.2.0                   py27_0  
simplegeneric             0.8.1                    py27_1  
singledispatch            3.4.0.3                  py27_0  
sip                       4.18                     py27_0  
six                       1.10.0                   py27_0  
snowballstemmer           1.2.1                    py27_0  
sockjs-tornado            1.0.3                    py27_0  
sphinx                    1.4.8                    py27_0  
spyder                    3.0.1                    py27_0  
sqlalchemy                1.1.3                    py27_0  
sqlite                    3.13.0                        0  
ssl_match_hostname        3.4.0.2                  py27_1  
statsmodels               0.6.1               np111py27_1  
sympy                     1.0                      py27_0  
system                    5.8                           2  
terminado                 0.6                      py27_0  
tk                        8.5.18                        0  
toolz                     0.8.0                    py27_0  
tornado                   4.4.2                    py27_0  
traitlets                 4.3.1                    py27_0  
traits                    4.5.0                    py27_0  
traitsui                  4.5.1                    py27_0  
unicodecsv                0.14.1                   py27_0  
vlfeat                    0.9.20                        1    menpo
vtk                       7.0.0                    py27_0    menpo
wcwidth                   0.1.7                    py27_0  
werkzeug                  0.11.11                  py27_0  
wheel                     0.29.0                   py27_0  
widgetsnbextension        1.2.6                    py27_0  
wrapt                     1.10.8                   py27_0  
xlrd                      1.0.0                    py27_0  
xlsxwriter                0.9.3                    py27_0  
xlwt                      1.1.2                    py27_0  
xz                        5.2.2                         0  
yaml                      0.1.6                         0  
zeromq                    4.1.5                         0  
zlib                      1.2.8                         3  
```

**Comment by liamwhan** [comment]
+1 for this error 

**EDIT:** I've just noticed something in the stack trace that's more than likely the culprit. For some reason the `traitlets` and `ipwidgets` packages are getting called from **outside** the conda environment. Note the difference between the 1st and second directories in the stack trace excerpt below:
```
###Omitted###\Anaconda3\envs\menpo\lib\site-packages\menpowidgets\style.py in format_box(box, box_style, border_visible, border_colour, border_style, border_width, border_radius, padding, margin)
     84         The margin around the widget.
     85     """
---> 86     box.box_style = box_style
     87     box.padding = padding
     88     box.margin = margin

###Omitted###\AppData\Roaming\Python\Python35\site-packages\traitlets\traitlets.py in __set__(self, obj, value)
    581             raise TraitError('The "%s" trait is read-only.' % self.name)
    582         else:
--> 583             self.set(obj, value)
    584 
    585     def _validate(self, obj, value):
```
**/EDIT**

my conda list output below: 

```
apptools                  4.4.0                    py35_0    menpo
backports                 1.0                      py35_0
boost                     1.59.0                   py35_0    menpo
bzip2                     1.0.6                    vc14_3  [vc14]
configobj                 5.0.6                    py35_0
cycler                    0.10.0                   py35_0
cyffld2                   0.2.4                    py35_0    menpo
cypico                    0.2.7                    py35_0    menpo
cyrasterize               0.3.2                    py35_0    menpo
cyvlfeat                  0.4.5                    py35_0    menpo
decorator                 4.0.10                   py35_1
dlib                      18.18                    py35_2    menpo
docopt                    0.6.2                    py35_0
entrypoints               0.2.2                    py35_0
envisage                  4.5.1                    py35_0    menpo
ffmpeg                    2.7.0                         0    menpo
fftw                      3.3.4                         0    menpo
freetype                  2.5.5                    vc14_1  [vc14]
get_terminal_size         1.0.0                    py35_0
glew                      2.0.0                    vc14_0  [vc14]  menpo
glfw3                     3.2.1                    vc14_0  [vc14]  menpo
hdf5                      1.8.17                   vc14_0  [vc14]
iconv                     1.14                          0    menpo
icu                       57.1                     vc14_0  [vc14]
ipykernel                 4.5.2                    py35_0
ipython                   4.2.0                    py35_0
ipython_genutils          0.1.0                    py35_0
ipywidgets                4.1.1                    py35_0
jinja2                    2.8                      py35_1
jpeg                      8d                       vc14_2  [vc14]
jsonschema                2.5.1                    py35_0
jupyter                   1.0.0                    py35_3
jupyter_client            4.4.0                    py35_0
jupyter_console           5.0.0                    py35_0
jupyter_core              4.2.1                    py35_0
libpng                    1.6.22                   vc14_0  [vc14]
libtiff                   4.0.6                    vc14_2  [vc14]
libxml2                   2.7.8                         0    menpo
markupsafe                0.23                     py35_2
matplotlib                1.5.3               np111py35_1
mayavi                    4.5.0                    py35_0    menpo
menpo                     0.7.6                    py35_0    menpo
menpo3d                   0.4.2                    py35_0    menpo
menpocli                  0.0.2                    py35_0    menpo
menpodetect               0.4.0                    py35_0    menpo
menpofit                  0.4.1                    py35_0    menpo
menpoproject              2.0                        py_0    menpo
menpowidgets              0.2.1                    py35_0    menpo
mistune                   0.7.3                    py35_0
mkl                       11.3.3                        1
mock                      2.0.0                    py35_0
nbconvert                 4.2.0                    py35_0
nbformat                  4.2.0                    py35_0
nose                      1.3.7                    py35_1
notebook                  4.3.0                    py35_0
numpy                     1.11.2                   py35_0
opencv3                   3.1.0                    py35_0    menpo
openssl                   1.0.2j                   vc14_0  [vc14]
pandas                    0.19.1              np111py35_0
path.py                   8.2.1                    py35_0
pbr                       1.10.0                   py35_0
pickleshare               0.7.4                    py35_0
pillow                    3.4.2                    py35_0
pip                       9.0.1                    py35_1
prompt_toolkit            1.0.9                    py35_0
pyface                    5.1.0                    py35_0    menpo
pygments                  2.1.3                    py35_0
pyparsing                 2.1.4                    py35_0
pyqt                      5.6.0                    py35_1
pyreadline                2.1                      py35_0
python                    3.5.2                         0
python-dateutil           2.6.0                    py35_0
pytz                      2016.10                  py35_0
pyzmq                     16.0.2                   py35_0
qt                        5.6.2                    vc14_0  [vc14]
qtconsole                 4.2.1                    py35_2
scikit-learn              0.18.1              np111py35_0
scipy                     0.18.1              np111py35_0
setuptools                27.2.0                   py35_1
simplegeneric             0.8.1                    py35_1
sip                       4.18                     py35_0
six                       1.10.0                   py35_0
tk                        8.5.18                   vc14_0  [vc14]
tornado                   4.4.2                    py35_0
traitlets                 4.3.1                    py35_0
traits                    4.5.0                    py35_0    menpo
traitsui                  5.1.0                    py35_0    menpo
vcomp                     1.0                      vc14_0  [vc14]  menpo
vlfeat                    0.9.20                   vc14_1  [vc14]  menpo
vs2015_runtime            14.0.25123                    0
vtk                       7.0.0                    py35_0    menpo
wcwidth                   0.1.7                    py35_0
wheel                     0.29.0                   py35_0
zlib                      1.2.8                    vc14_3  [vc14]

```

**Comment by nontas** [comment]
I'm not sure why this is happening to you.... Your conda list looks fine. I just made a fresh installation of `menpowidgets` in a new conda environment and everything works as expected - I can't reproduce the error. Can you check that as well?

```
conda create -n test python=3.5
conda install -c menpo menpowidgets
```


In general the `box_style` error has to do with `ipywidgets=5.*`. It cannot get a `None` value anymore. But it should be fine with `ipywidgets=4.1.1`. Keep in mind that a big upgrade and new release is coming soon (https://github.com/menpo/menpowidgets/pull/22) which will make `menpowidgets` compatible with the latest `ipywidgets`.

**Comment by liamwhan** [comment]
I just resolved this error. 

I suspect this issue is unique to Windows with side by side Python and Anaconda installs, I'm running Win10 x64 (v10.0.14393) and have Python 3.5 installed alongside Anaconda (which is NOT recommended by Anaconda installer so this is likely the reason. Slap on the wrist for me.)

The problem seems to be that if you have Python installed alongside Conda, and there is a newer version of `traitlets` or `ipywidgets` installed in your base Python `site-packages` then the package loader will preference the newer version over the environments older packages. 

This persisted when I used the Menpo Playground, even when running with the python.cmd supplied it loaded `traitlets` from `%USERPROFILE%/AppData/Roaming/Python/Python35/site-packages`.

Unsetting PYTHONPATH and any PATH references to `%USERPROFILE%/AppData/Roaming/Python/Python35/site-packages` had no effect. 

Once I removed `ipywidgets` and `traitlets` packages from the `%USERPROFILE%/AppData/Roaming/Python/Python35/site-packages` the problem was resolved.
